src/overlays/actors/ovl_Obj_Kibako2/z_obj_kibako2.c

```diff
@@ -31,21 +33,143 @@ const ActorInit Obj_Kibako2_InitVars = {
     (ActorFunc)ObjKibako2_Update,
     (ActorFunc)ObjKibako2_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Kibako2/func_80B959D0.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Kibako2/func_80B95A28.s")
+static ColliderCylinderInit sCylinderInit = {
+    { COLTYPE_UNK10, 0x00, 0x09, 0x00, 0x20, COLSHAPE_CYLINDER },
+    { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x40000040, 0x00, 0x00 }, 0x00, 0x01, 0x00 },
+    { 31, 48, 0, { 0, 0, 0 } },
+};
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 3000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 500, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 1000, ICHAIN_STOP),
+};
+
+void ObjKibako2_InitCollider(Actor* thisx, GlobalContext* globalCtx) {
+    ObjKibako2* this = THIS;
+
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->dyna.actor, &sCylinderInit);
+    Collider_CylinderUpdate(&this->dyna.actor, &this->collider);
+}
+
+void ObjKibako2_Break(ObjKibako2* this, GlobalContext* globalCtx) {
+    s32 padE4;
+    s32 padE0;
+    PosRot* thisPosRot;
```

Comment:
ive never seen a temp for PosRot actually be real before-- this may hint at a temp for `thisx`perhaps. less likely would be for the position vector specifically, but it happens sometimes

Reply:
It's definitely not a temp for `thisx`; that causes some pretty weird instruction differences. But a temp pointer to the position vector definitely works, so let's go with that for now.